pact/provider/pact-broker-loader/src/main/java/org/arquillian/algeron/pact/provider/core/loader/pactbroker/PactBrokerLoader.java

```diff
@@ -59,36 +58,32 @@
             .build();
     private Callable<HttpResponse> httpResponseCallable;
 
-    public PactBrokerLoader(final String pactBrokerHost, final String pactBrokerPort, final String pactBrokerProtocol) {
-        this(pactBrokerHost, pactBrokerPort, pactBrokerProtocol, Collections.singletonList("latest"));
+    public PactBrokerLoader() {
+        super();
     }
 
-    public PactBrokerLoader(final String pactBrokerHost, final String pactBrokerPort, final String pactBrokerProtocol,
-                            final List<String> tags) {
-        this.pactBrokerHost = pactBrokerHost;
-        this.pactBrokerPort = pactBrokerPort;
-        this.pactBrokerProtocol = pactBrokerProtocol;
-        this.pactBrokerTags = tags;
+
+    @Override
+    public void setProviderName(String providerName) {
+        this.providerName = providerName;
     }
 
-    public PactBrokerLoader(final PactBroker pactBroker) {
-        this(getResolvedValue(pactBroker.host()),
-                getResolvedValue(pactBroker.port()),
-                getResolvedValue(pactBroker.protocol()),
-                        Arrays.stream(pactBroker.tags())
-                                .map(tag -> getResolvedValue(tag))
-                                .collect(toList()));
+    @Override
+    public void configure(Map<String, Object> configuration) {
+        this.pactBroker = new PactBrokerImpl(configuration);
     }
 
     @Override
-    public void setProviderName(String providerName) {
-        this.providerName = providerName;
+    public String getName() {
+        return "pactbroker";
     }
 
     @Override
     public List<URI> retrieve() throws IOException {
         List<URI> pacts = new ArrayList<>();
-        for (String tag: pactBrokerTags) {
+        for (String tag: Arrays.stream(pactBroker.tags())
```

Comment:
Regular `for` mixed with `stream` looks a bit odd for my old eyes. Can't we use our own [`Collector`](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collector.html) instead?